src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1

```diff
@@ -355,6 +363,50 @@ class VariableGroup: ADOSVTBase
         return $controlResult;
     }
 
+    hidden [ControlResult] CheckCredInVarGrpAutomatedFix([ControlResult] $controlResult){
+        try{
+            $RawDataObjForControlFix = @();
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject
+            $varList = @();
+            if (-not $this.UndoFix) {
+                $RawDataObjForControlFix.PSObject.Properties | foreach {
+                    #The api does not allow updating individual variables inside a var grp, all variables have to be a part of the body or else they will be removed from the grp.
+                    #Hence using the global var grp object to store all variables details inside the post body and updating only the required variable.
+                    $this.VarGrp.variables.($_.Name) | Add-Member NoteProperty -name "isSecret" -value $true                    
+                    $varList+=$_.Name;
+                    
+                }
+                $controlResult.AddMessage([VerificationResult]::Fixed,  "Following variables have been marked as secret: ");
+               
+            }
+            else {
+                $RawDataObjForControlFix.PSObject.Properties | foreach {  
+                    #The api does not allow updating individual variables inside a var grp, all variables have to be a part of the body or else they will be removed from the grp.
```

Comment:
Does this mean raw object will have all variables data (both with and without credentials)?

Reply:
No, raw object will only have variables data with credentials (in encrypted format). The rest of the variables are part of  the global var grp object formed in the constructor, we just reuse this object and update the required variables inside this obj